src/ng-select/ng-dropdown-panel.component.ts

```diff
@@ -1,35 +1,35 @@
+import { DOCUMENT } from '@angular/common';
 import {
+    ChangeDetectionStrategy,
     Component,
-    OnDestroy,
-    Renderer2,
     ElementRef,
-    Input,
     EventEmitter,
-    Output,
-    ViewChild,
-    SimpleChanges,
+    HostListener,
+    Inject,
+    Input,
     NgZone,
-    TemplateRef,
-    ViewEncapsulation,
-    ChangeDetectionStrategy,
-    AfterContentInit,
-    OnInit,
     OnChanges,
-    HostListener,
+    OnDestroy,
+    OnInit,
     Optional,
-    Inject
+    Output,
+    Renderer2,
+    SimpleChanges,
+    TemplateRef,
+    ViewChild,
+    ViewEncapsulation
 } from '@angular/core';
-import { DOCUMENT } from '@angular/common';
+import { animationFrameScheduler, asapScheduler, fromEvent, merge, Subject } from 'rxjs';
+import { auditTime, takeUntil } from 'rxjs/operators';
+import { NgDropdownPanelService, PanelDimensions } from './ng-dropdown-panel.service';
 
-import { NgOption } from './ng-select.types';
 import { DropdownPosition } from './ng-select.component';
-import { WindowService } from './window.service';
-import { VirtualScrollService } from './virtual-scroll.service';
-import { takeUntil } from 'rxjs/operators';
-import { Subject, fromEvent, merge } from 'rxjs';
+import { NgOption } from './ng-select.types';
+import { isDefined } from './value-utils';
 
 const TOP_CSS_CLASS = 'ng-select-top';
 const BOTTOM_CSS_CLASS = 'ng-select-bottom';
+const SCROLL_SCHEDULER = typeof requestAnimationFrame !== 'undefined' ? animationFrameScheduler : asapScheduler;
```

Comment:
From what I see reaquestAnimationFrame is already supported even on IE10 https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame

In which case can it be undefined?

Reply:
while using server-side rendering (angular universal) requestAnimationFrame will be undefined